Applications/Spire/Source/OrderImbalanceIndicator/LocalOrderImbalanceIndicatorModel.cpp

```diff
@@ -1 +1,78 @@
 #include "Spire/OrderImbalanceIndicator/LocalOrderImbalanceIndicatorModel.hpp"
+#include "Spire/Spire/Utility.hpp"
+
+using namespace Nexus;
+using namespace Spire;
+
+void LocalOrderImbalanceIndicatorModel::publish(
+    const OrderImbalance& imbalance) {
+  insert_sorted(imbalance);
+  m_imbalance_published_signal(imbalance);
+}
+
+void LocalOrderImbalanceIndicatorModel::insert(
+    const OrderImbalance& imbalance) {
+  insert_sorted(imbalance);
+}
+
+QtPromise<std::vector<Nexus::OrderImbalance>>
+    LocalOrderImbalanceIndicatorModel::load(const TimeInterval& interval) {
+  auto first = std::vector<OrderImbalance>::iterator();
+  auto last = std::vector<OrderImbalance>::iterator();
+  if(boost::icl::is_left_closed(interval.bounds())) {
```

Comment:
Add a ```using namespace boost;``` and ```using namespace boost::icl;```. You can/should do that with every namespace other than ```std```.

Reply:
Should I leave it on the boost::optional because we sometimes mixed std and boost optionals?

Reply:
No because we never do ```using namespace std;``` so there is never a conflict.